stl/inc/format

```diff
@@ -1571,8 +1506,14 @@ public:
     }
 
 private:
-    size_t _Num_args              = 0;
-    const unsigned char* _Storage = nullptr;
+    template <class _Ty>
+    _NODISCARD static auto _Get_value_from_memory(const unsigned char* _Val) noexcept {
+        auto& _Temp = *reinterpret_cast<const unsigned char(*)[sizeof(_Ty)]>(_Val);
+        return _Bit_cast<_Ty>(_Temp);
+    }
+
+    size_t _Num_args                      = 0;
+    const _Format_arg_index* _Index_array = nullptr;

```

Comment:
at some point we should add extensive comments around how this all works and what the storage layout is, otherwise we'll all forget. It's not super obvious code.

Reply:
There's still no comment here.

Just "Note this is actually a pointer to _Format_arg_store" or something (we could just change to to actually be a pointer to _Format_arg_store, or put the two storage fields in a structure and store a pointer to that here, and a value of that type in _Format_arg_store.

Reply:
There's no comment here, but there's a comment at the usage in `get` directing us to the long explanatory comment in `_Format_arg_store`.

I suppose we could make this a `void*` that points to the `_Format_arg_store` (we can't use the proper type because erasing the type is the point), it's all the same address, after all. Doing so would introduce an additional `static_cast` to convert the `void*` to `_Format_arg_index*` in `get`, and I figured `get` was already complicated enough.

`_Format_arg_store` is already "the two storage fields in a structure".

Would you like an additional comment here directing the reader to the comment in `_Format_arg_store`? How about "This points to the first element of the member array `_Index_array` in an instance of `_Format_arg_store<_Num_args>`"? Or would you prefer this to be a `const void*` that "points to an instance of `_Format_arg_store<_Num_args>`"?

Reply:
no, this is fine for now. We could have the actual storage in another struct that didn't depend on the argument types / number of arguments, but it's not a big deal.

Reply:
since I seem to be one one who cares I should probably be the one to make the change anyways